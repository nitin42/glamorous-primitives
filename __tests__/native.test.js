import React from 'react'
import {render, mount} from 'enzyme'
import serializer from 'jest-glamor-react'
import 'react-primitives';
import renderer from 'react-test-renderer'
import reactPrimitives, {StyleSheet, View} from 'react-primitives'
import glamorous from '../src/index'

expect.addSnapshotSerializer(serializer)

describe('Primitive interfaces', () => {
  test('sanity test', () => {
    const Component = glamorous.text({margin: '10px'})

    const tree = renderer.create(
      <Component>Hello World</Component>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should not throw any error when called glamorous.primitive()', () => {
    glamorous.view({})
  })

  test('should combine with StyleSheet.create() (inline styles)', () => {
    const Component = glamorous.view({
      margin: '10px',
      justifyContent: 'center',
      alignItems: 'center',
    })

    const styles = StyleSheet.create({
      foo: {
        backgroundColor: 'red',
        display: 'flex',
      },
    })

    const tree = renderer.create(
      <Component style={styles.foo} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should attach a displayName', () => {
    const ViewComponent = glamorous.view({})
    const ImageComponent = glamorous.image({})
    const TextComponent = glamorous.text({})
    const TouchableComponent = glamorous.touchable({})

    expect(ViewComponent.displayName).toBe('glamorous(View)')
    expect(ImageComponent.displayName).toBe('glamorous(Image)')
    expect(TextComponent.displayName).toBe('glamorous(Text)')
    expect(TouchableComponent.displayName).toBe('glamorous(Touchable)')
  })

  test('should pass the ref to the component', () => {
    const Component = glamorous.view({})
    const ref = jest.fn()

    const wrapper = mount(<Component innerRef={ref} />)
    const View = wrapper.find('View').first()

    // ReactWrapper:getNode() is no longer supported so use ReactWrapper:instance()
    expect(ref).toHaveBeenCalledWith(View.instance())
  })

  test('should render a component with theme properties', () => {
    const Component = glamorous.text(
      {
        color: 'red',
      },
      (props, theme) => ({padding: theme.padding}),
    )
    expect(
      render(<Component theme={{padding: '10px'}} />),
    ).toMatchSnapshot()
  })

  test('should pass an updated theme when theme prop changes', () => {
    const Component = glamorous.text(
      {
        color: 'red',
      },
      (props, theme) => ({padding: theme.padding}),
    )
    const wrapper = mount(<Component theme={{padding: 10}} />)
    expect(wrapper).toMatchSnapshot(
      `with theme prop of padding 10px`,
    )
    wrapper.setProps({theme: {padding: 20}})
    expect(wrapper).toMatchSnapshot(
      `with theme prop of padding 20px`,
    )
  })

  test('should render the <Image /> component', () => {
    const ReactImage = glamorous.image({padding: '10px'})
    expect(
      <ReactImage source="https://facebook.github.io/react/img/logo_og.png" />,
    ).toMatchSnapshot()
  })

  test('combine inline styles with <Image /> component', () => {
    const styles = StyleSheet.create({
      image: {
        justifyContent: 'center',
        alignItems: 'center',
      },
    })
    const ReactImage = glamorous.image({padding: '10px'})
    expect(
      <ReactImage
        style={styles.image}
        source="https://facebook.github.io/react/img/logo_og.png"
      />,
    ).toMatchSnapshot()
  })

  test('supports props as css properties', () => {
    const tree = renderer.create(
      <glamorous.Text color="pink">Hello World</glamorous.Text>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
