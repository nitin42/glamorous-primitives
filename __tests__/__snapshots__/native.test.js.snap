// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primitive interfaces combine inline styles with <Image /> component 1`] = `
<glamorous(Image)
  source="https://facebook.github.io/react/img/logo_og.png"
  style={79}
/>
`;

exports[`Primitive interfaces sanity test 1`] = `
<div
  className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-156q2ks rn-marginRight-zso239 rn-marginBottom-15d164r rn-marginLeft-1n0xq6e rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
>
  Hello World
</div>
`;

exports[`Primitive interfaces should combine with StyleSheet.create() (inline styles) 1`] = `
<div
  className="rn-alignItems-1awozwy rn-backgroundColor-1mjtqww rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-156q2ks rn-marginRight-zso239 rn-marginBottom-15d164r rn-marginLeft-1n0xq6e rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
/>
`;

exports[`Primitive interfaces should pass an updated theme when theme prop changes: with theme prop of padding 10px 1`] = `
<glamorous(Text)
  theme={
    Object {
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        77,
        Object {
          "padding": 10,
        },
      ]
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-1lbce71 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "paddingBottom": "10px",
          "paddingLeft": "10px",
          "paddingRight": "10px",
          "paddingTop": "10px",
        }
      }
    />
  </Text>
</glamorous(Text)>
`;

exports[`Primitive interfaces should pass an updated theme when theme prop changes: with theme prop of padding 20px 1`] = `
<glamorous(Text)
  theme={
    Object {
      "padding": 20,
    }
  }
>
  <Text
    style={
      Array [
        77,
        Object {
          "padding": 20,
        },
      ]
    }
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-1lbce71 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "paddingBottom": "20px",
          "paddingLeft": "20px",
          "paddingRight": "20px",
          "paddingTop": "20px",
        }
      }
    />
  </Text>
</glamorous(Text)>
`;

exports[`Primitive interfaces should render a component with theme properties 1`] = `
<div
  class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-1lbce71 rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style="padding-top:10px;padding-right:10px;padding-bottom:10px;padding-left:10px"
/>
`;

exports[`Primitive interfaces should render the <Image /> component 1`] = `
<glamorous(Image)
  source="https://facebook.github.io/react/img/logo_og.png"
/>
`;

exports[`Primitive interfaces supports props as css properties 1`] = `
<div
  className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
  dir="auto"
  style={
    Object {
      "color": "pink",
    }
  }
>
  Hello World
</div>
`;
